sympy/functions/elementary/trigonometric.py

```diff
@@ -2677,6 +2677,8 @@ def taylor_term(n, x, *previous_terms):
     def _eval_as_leading_term(self, x, logx=None, cdir=0):  # atan
         arg = self.args[0]
         x0 = arg.subs(x, 0).cancel()
+        if x0 is S.NaN:
+            raise PoleError()
         if x0.is_zero:
```

Comment:
Also another thing being we would encounter this case in all inverse trigonometric functions (while calculating limits for 
`asin(exp(x))`, `acsc(exp(x))`). Just addressing an individual case might not help us overall, could you try tackling all cases that go wrong.

Reply:
`nan` typically results when a substitution gives 0/0 (e.g. `arg = sin(x)/x`). It may be advisable to first try to compute `x0` as a limit before raising `PoleError`.

Reply:
e.g. `limit(acos(sin(x)/x),x,oo)` ? @borgesaugusto, what does that do in this PR?

Reply:
This: limit(acos(sin(x)),x,oo) I*log(I*AccumBounds(-1, 1) + AccumBounds(0, 1)) + pi/2
and 
limit(acos(sin(x)/x),x,oo) gives "sympy.core.function.PoleError: Cannot expand sin(1/_w) around 0" in "sympy/functions/elementary/trigonometric.py" line 436" (The txt with the trace), ending with a NotImplementedError.
[sympy_trace.txt](https://github.com/sympy/sympy/files/12840213/sympy_trace.txt)